twilight-gateway/src/shard.rs

```diff
@@ -873,11 +879,9 @@ impl Shard {
             Ok(OpCode::Heartbeat) => {
                 let event = Self::parse_event(buffer)?;
 
-                if let Err(source) = self.heartbeat(Some(event.data)).await {
-                    self.disconnect(Disconnect::Resume);
-
-                    return Err(ProcessError::from_send(source));
-                }
+                self.heartbeat(Some(event.data))
+                    .await
+                    .map_err(ProcessError::from_send)?;
```

Comment:
we should still disconnect when we can't heartbeat, as a more careful measure.

Reply:
The heartbeat method only errors on sending the message, which already disconnects the shard, so explicitly calling disconnect is not needed